blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DPreaggregatedMetricsRW.java

```diff
@@ -0,0 +1,349 @@
+
+                    // add all ResultSetFutures for a particular locator together
+                    List<ResultSetFuture> existing = locatorToFuturesMap.get(locator);
+                    if (existing == null) {
+                        existing = new ArrayList<ResultSetFuture>();
+                        locatorToFuturesMap.put(locator, existing);
+                    }
+                    existing.addAll(selectFutures);
+
+                } catch (CacheException ex) {
+                    LOG.error(String.format("Error looking up locator %s in cache", locator), ex);
+                }
+            }
+            return resultSetsToMetricData(locatorToFuturesMap, locatorRollupTypeMap, granularity);
+        } finally {
+            ctx.stop();
+        }
+    }
+
+    /**
+     * Converts a list of {@link com.datastax.driver.core.ResultSetFuture} for each
+     * {@link com.rackspacecloud.blueflood.types.Locator} to
+     * {@link com.rackspacecloud.blueflood.outputs.formats.MetricData} object.
+     *
+     * @param resultSets
+     * @param locatorRollupType
+     * @param granularity
+     * @return
+     */
+    protected Map<Locator, MetricData> resultSetsToMetricData(Map<Locator,
+                                                              List<ResultSetFuture>> resultSets,
+                                                              Map<Locator, RollupType> locatorRollupType,
+                                                              Granularity granularity) {
+
+        // iterate through all ResultSetFuture
+        Map<Locator, MetricData> locatorMetricDataMap = new HashMap<Locator, MetricData>();
+        for (Map.Entry<Locator, List<ResultSetFuture>> entry : resultSets.entrySet() ) {
+            Locator locator = entry.getKey();
+            List<ResultSetFuture> futures = entry.getValue();
+
+            try {
+                RollupType rollupType = locatorRollupType.get(locator);
+
+                DAbstractPreaggregatedIO io = rollupTypeToIO.get(rollupType);
+
+                // get ResultSets to a Table of locator, timestamp, rollup
+                Table<Locator, Long, Rollup> locatorTimestampRollup = io.toLocatorTimestampRollup(futures, locator, granularity);
+
+                // convert to Points and MetricData
+                Points<Rollup> points = convertToPoints(locatorTimestampRollup.row(locator));
+
+                // get the dataType for this locator
+                DataType dataType = getDataType(locator, MetricMetadata.TYPE.name().toLowerCase());
+
+                // create MetricData
+                MetricData.Type outputType = MetricData.Type.from(rollupType, dataType);
+                MetricData metricData = new MetricData(points, getUnitString(locator), outputType);
+                locatorMetricDataMap.put(locator, metricData);
+
+            } catch (Exception ex) {
```

Comment:
I was incorrect, as you point out. The toLocatorTimestampRollup() has its own try-catch. But I will change this ```catch(Exception ex)``` to ```catch(CacheException ex)``` as the ```getDataType()``` throws it.